File: 1247-minimum-swaps-to-make-strings-equal/1247-minimum-swaps-to-make-strings-equal.py
class Solution:
    def minimumSwap(self, s1: str, s2: str) -> int:
        x = y = 0
        for c1, c2 in zip(s1, s2):
            if c1 == c2:
                continue
            if c1 == 'x':
                x += 1
            else:
                y += 1
        return -1 if (x + y) & 1 else (x + 1) // 2 + (y + 1) // 2 
                    

File: 1247-minimum-swaps-to-make-strings-equal/1247-minimum-swaps-to-make-strings-equal.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {number}
 */
var minimumSwap = function(s1, s2) {
  let [x, y] = [0, 0];
  
  for (let i = 0; i < s1.length; i++) {
    if (s1[i] === s2[i]) continue;
    s1[i] === 'x' ? x++ : y++;
  }
  
  return (x + y) & 1 ? -1 : ~~((x + 1) / 2) + ~~((y + 1) / 2);
};